modules/nlu/src/backend/storage.ts

```diff
@@ -168,14 +177,16 @@ export default class Storage {
     return this.ghost.upsertFile(this.modelsDir, modelName, modelBuffer)
   }
 
-  async getAvailableModels(): Promise<AvailableModel[]> {
+  async getAvailableModels(): Promise<ModelMeta[]> {
     const models = await this.ghost.directoryListing(this.modelsDir, '*.bin')
     return models.map(x => {
-      const fileName = path.basename(x, '.bin')
-      const parts = fileName.split('__')
-      return <AvailableModel>{
+      const fileName = path.basename(x)
+      const parts = fileName.replace('.bin', '').split('__')
+      return {
+        fileName,
         created_on: new Date(parts[0]),
-        hash: parts[1]
+        hash: parts[1],
```

Comment:
will crash if there's a file named `yolo__.bin` ?

Reply:
nop, wont crash! will return undefined which in our case would cause a retraining